core/src/main/java/com/capitalone/dashboard/model/Cmdb.java

```diff
@@ -193,68 +192,28 @@ public void setValidConfigItem(boolean validConfigItem) {
 
     public List<String> getComponents(){ return components; }
 
-    @Override
-    public int hashCode(){
-        return Objects.hash(configurationItem,
-                assignmentGroup,
-                appServiceOwner,
-                businessOwner,
-                supportOwner,
-                developmentOwner,
-                ownerDept,
-                itemType,
-                configurationItemSubType,
-                configurationItemType);
-    }
-    @Override
-    public boolean equals(Object compareTo){
+    public boolean isCmdbMatch(Object compareTo){
         boolean doesEqual = true;
 
         if(compareTo == null || !compareTo.getClass().isAssignableFrom(Cmdb.class)){
             doesEqual = false;
         }else {
             Cmdb newCmdb = (Cmdb) compareTo;
 
-            if(!newCmdb.toString().equals(toString())){
+            if((newCmdb.getConfigurationItem() !=null && !newCmdb.getConfigurationItem().equals(configurationItem)) ||
+                    (newCmdb.getAssignmentGroup() !=null && !newCmdb.getAssignmentGroup().equals(assignmentGroup)) ||
+                    (newCmdb.getAppServiceOwner() !=null && !newCmdb.getAppServiceOwner().equals(appServiceOwner)) ||
+                    (newCmdb.getBusinessOwner() !=null && !newCmdb.getBusinessOwner().equals(businessOwner)) ||
+                    (newCmdb.getSupportOwner() !=null && !newCmdb.getSupportOwner().equals(supportOwner)) ||
+                    (newCmdb.getDevelopmentOwner() !=null && !newCmdb.getDevelopmentOwner().equals(developmentOwner)) ||
+                    (newCmdb.getOwnerDept() !=null && !newCmdb.getOwnerDept().equals(ownerDept)) ||
+                    (newCmdb.getConfigurationItemSubType() !=null && !newCmdb.getConfigurationItemSubType().equals(configurationItemSubType)) ||
+                    (newCmdb.getConfigurationItemType() !=null && !newCmdb.getConfigurationItemType().equals(configurationItemType)) ||
+                    (newCmdb.isValidConfigItem() != validConfigItem)){
                 doesEqual = false;
             }
         }
 
```

Comment:
If these are all strings, have an utility method to check for null and equality. This looks little too error prone.